pkg/etcd/etcd.go

```diff
@@ -1116,6 +1158,47 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	return e.setLearnerProgress(ctx, progress)
 }
 
+func (e *ETCD) setEtcdStatusCondition(node *v1.Node, memberName string, promoted bool) (*v1.Node, error) {
+	etcdStatusType := v1.NodeConditionType("EtcdIsVoter")
+
+	var newCondition v1.NodeCondition
+	if promoted {
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "True",
+			Reason:  "MemberNotLearner",
+			Message: "Node is a voting member of the etcd cluster",
+		}
+	} else {
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "MemberIsLearner",
+			Message: "Node has not been promoted to voting member of the etcd cluster",
+		}
+	}
+
+	if find, condition := nodeUtil.GetNodeCondition(&node.Status, etcdStatusType); find >= 0 {
+		if condition.Status == newCondition.Status {
+			logrus.Debugf("Member %s is not changing etcd status condition", memberName)
+			condition.LastHeartbeatTime = metav1.Now()
+			return e.config.Runtime.Core.Core().V1().Node().UpdateStatus(node)
```

Comment:
I think we should use the status patch helper like the kubelet does; the status itself may be large and re-sending the whole thing instead of just patching is likely to conflict given how many other statuses are also being updated regularly.

https://github.com/kubernetes/kubernetes/blob/66e20ff891c6d3a5da75cb90415055bd1fece07d/pkg/kubelet/kubelet_node_status.go#L632-L637